azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -49,13 +61,36 @@
 
   private final ImageTypeDao imageTypeDao;
 
-  private static String INSERT_IMAGE_VERSION_QUERY =
-      "insert into image_versions ( path, description, "
-          + "version, type_id, state, release_tag, created_by, modified_by) "
-          + "values (?, ?, ?, ?, ?, ?, ?, ?)";
-  private static String SELECT_IMAGE_VERSION_BASE_QUERY = "select iv.id, iv.path, iv.description, "
-      + "iv.version, it.name, iv.state, iv.release_tag, iv.created_on, iv.created_by, iv.modified_on, "
-      + "iv.modified_by from image_versions iv, image_types it where it.id = iv.type_id";
+  private static final String INSERT_IMAGE_VERSION_QUERY =
+      "insert into image_versions ( path, description, version, type_id, state, release_tag, "
+          + "created_by, created_on, modified_by, modified_on) "
+          + "values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+  private static final String SELECT_IMAGE_VERSION_BASE_QUERY =
+      "select iv.id, iv.path, iv.description, "
+          + "iv.version, cast(replace(iv.version, '.', '') as unsigned integer) as int_version, "
+          + "it.name, iv.state, iv.release_tag, iv.created_on, iv.created_by, iv.modified_on, "
+          + "iv.modified_by from image_versions iv, image_types it where it.id = iv.type_id";
+  /**
+   * The below query uses calculated column to get the image version in integer format. The query
+   * consists for two inner queries. One inner query selects the max image version using the
+   * calculated int_version field for the image types. The second inner query matches the max
+   * version computed above and selects the corresponding image version records. NOTE: The computed
+   * column can't be used directly in the where clause hence, two inner tables are created on top of
+   * the inner queries.
+   */
+  private static final String SELECT_LATEST_ACTIVE_IMAGE_VERSION_QUERY = "select outer_tbl.id, "
+      + "outer_tbl.path, outer_tbl.description, outer_tbl.version, outer_tbl.name, outer_tbl.state, "
```

Comment:
It would be much better for readability and maintainability to separate out inner queries in separate strings and then combine them together.